crates/sui-core/src/unit_tests/authority_tests.rs

```diff
@@ -4909,3 +4965,129 @@ fn test_choose_next_system_packages() {
+    fn make_gas_coins(owner: SuiAddress, gas_amount: u64, coin_num: u64) -> Vec<Object> {
+        let mut objects = vec![];
+        let coin_balance = gas_amount / coin_num;
+        for _ in 1..coin_num {
+            let gas_object_id = ObjectID::random();
+            objects.push(Object::with_id_owner_gas_for_testing(
+                gas_object_id,
+                owner,
+                coin_balance,
+            ));
+        }
+        // in case integer division dropped something, make a coin with whatever is left
+        let amount_left = gas_amount - (coin_balance * (coin_num - 1));
+        let gas_object_id = ObjectID::random();
+        objects.push(Object::with_id_owner_gas_for_testing(
+            gas_object_id,
+            owner,
+            amount_left,
+        ));
+        objects
+    }
+
+    // run an object creation transaction with the given amount of gas and coins
+    async fn run_and_check_ok(reference_gas_used: u64, coin_num: u64, budget: u64) -> u64 {
+        let (sender, sender_key): (_, AccountKeyPair) = get_key_pair();
+        let gas_coins = make_gas_coins(sender, reference_gas_used, coin_num);
+        let gas_coin_ids: Vec<_> = gas_coins.iter().map(|obj| obj.id()).collect();
+        let (state, effects) = create_obj(sender, sender_key, gas_coins, budget).await;
+        // transaction is good
+        assert!(effects.status.is_ok());
+        // gas object in effects is first coin in vector of coins
+        assert_eq!(gas_coin_ids[0], effects.gas_object.0 .0);
+        // object is created and gas at position 0 mutated
+        assert_eq!((effects.created.len(), effects.mutated.len()), (1, 1));
+        // extra coin are deleted
+        assert_eq!(effects.deleted.len() as u64, coin_num - 1);
+        for gas_coin_id in &gas_coin_ids[1..] {
+            assert!(effects
+                .deleted
+                .iter()
+                .any(|deleted| deleted.0 == *gas_coin_id));
+        }
+        // balance on first coin is correct
+        let balance = sui_types::gas::get_gas_balance(
+            &state.get_object(&gas_coin_ids[0]).await.unwrap().unwrap(),
+        )
+        .unwrap();
+        let gas_used = effects.gas_cost_summary().gas_used();
+        assert!(reference_gas_used > balance);
+        assert_eq!(reference_gas_used, balance + gas_used);
+        gas_used
+    }
+
+    // run an object creation transaction with the given amount of gas and coins, failure case
+    async fn run_and_check_err(reference_gas_used: u64, coin_num: u64, budget: u64) -> u64 {
+        let (sender, sender_key): (_, AccountKeyPair) = get_key_pair();
+        let gas_coins = make_gas_coins(sender, reference_gas_used, coin_num);
+        let gas_coin_ids: Vec<_> = gas_coins.iter().map(|obj| obj.id()).collect();
+        let (state, effects) = create_obj(sender, sender_key, gas_coins, budget).await;
+        // transaction is good
+        assert!(effects.status.is_err());
```

Comment:
A bit lost. What is done to make this one a failure/error? Test looks _really_ similar to the one above

Reply:
I'll check